### ISSUE TITLE
npm ERR! Failed at the node-rdkafka@2.6.1 install script.

### ISSUE THREAD

**Issue by stackfortech** [Blizzard/node-rdkafka#619]
**Environment Information**
 - OS [e.g. Mac, Arch, Windows 10]: 
 - Node Version [e.g. 8.2.1]:
 - NPM Version [e.g. 5.4.2]:
 - C++ Toolchain [e.g. Visual Studio, llvm, g++]:
 - node-rdkafka version [e.g. 2.3.3]:

[2019-05-15T08:27:09.922Z] > node-rdkafka@2.6.1 install /home/jenkins/workspace/re-abm_mock-services_PR-172-2B4NNI45YR4WY3JG27ZJ2KJYL3FFWLZFKXTPDTNOATSMO6JZF3YQ/node_modules/node-rdkafka

[2019-05-15T08:27:09.922Z] > node-gyp rebuild

[2019-05-15T08:27:09.922Z] 

[2019-05-15T08:27:12.102Z] make: Entering directory '/home/jenkins/workspace/re-abm_mock-services_PR-172-2B4NNI45YR4WY3JG27ZJ2KJYL3FFWLZFKXTPDTNOATSMO6JZF3YQ/node_modules/node-rdkafka/build'

[2019-05-15T08:27:12.102Z]   ACTION deps_librdkafka_gyp_librdkafka_target_configure deps/librdkafka/config.h

[2019-05-15T08:27:12.102Z] env: can't execute 'bash': No such file or directory

[2019-05-15T08:27:12.102Z] Command failed: ./configure --prefix=/home/jenkins/workspace/re-abm_mock-services-2B4NNI45YR4WY3JG27ZJ2KJYL3FFWLZFKXTPDTNOATSMO6JZF3YQ/node_modules/node-rdkafka/build/deps --libdir=/home/jenkins/workspace/re-abm_mock-services_PR-172-2B4NNI45YR4WY3JG27ZJ2KJYL3FFWLZFKXTPDTNOATSMO6JZF3YQ/node_modules/node-rdkafka/build/deps

[2019-05-15T08:27:12.102Z] make: *** [deps/librdkafka.target.mk:13: deps/librdkafka/config.h] Error 1

[2019-05-15T08:27:12.102Z] make: Leaving directory '/home/jenkins/workspace/re-abm_mock-services_PR-172-2B4NNI45YR4WY3JG27ZJ2KJYL3FFWLZFKXTPDTNOATSMO6JZF3YQ/node_modules/node-rdkafka/build'

[2019-05-15T08:27:12.102Z] gyp ERR! build error 

[2019-05-15T08:27:12.102Z] gyp ERR! stack Error: `make` failed with exit code: 2

[2019-05-15T08:27:12.102Z] gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:258:23)

[2019-05-15T08:27:12.102Z] gyp ERR! stack     at emitTwo (events.js:126:13)

[2019-05-15T08:27:12.102Z] gyp ERR! stack     at ChildProcess.emit (events.js:214:7)

[2019-05-15T08:27:12.102Z] gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:198:12)

[2019-05-15T08:27:12.102Z] gyp ERR! System Linux 3.10.0-514.el7.x86_64

[2019-05-15T08:27:12.102Z] gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"

[2019-05-15T08:27:12.102Z] gyp ERR! cwd /home/jenkins/workspace/re-abm_mock-services_PR-172-2B4NNI45YR4WY3JG27ZJ2KJYL3FFWLZFKXTPDTNOATSMO6JZF3YQ/node_modules/node-rdkafka

[2019-05-15T08:27:12.102Z] gyp ERR! node -v v8.9.1

[2019-05-15T08:27:12.102Z] gyp ERR! node-gyp -v v3.6.2

[2019-05-15T08:27:12.102Z] gyp ERR! not ok 

[2019-05-15T08:27:12.775Z] npm WARN mock-services@0.0.1 No repository field.

[2019-05-15T08:27:12.775Z] 

[2019-05-15T08:27:12.775Z] npm ERR! code ELIFECYCLE

[2019-05-15T08:27:12.775Z] npm ERR! errno 1

[2019-05-15T08:27:12.775Z] npm ERR! node-rdkafka@2.6.1 install: `node-gyp rebuild`

[2019-05-15T08:27:12.775Z] npm ERR! Exit status 1

[2019-05-15T08:27:12.775Z] npm ERR! 

[2019-05-15T08:27:12.775Z] npm ERR! Failed at the node-rdkafka@2.6.1 install script.

[2019-05-15T08:27:12.775Z] npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

[2019-05-15T08:32:21.830Z] 

[2019-05-15T08:32:21.831Z] npm ERR! A complete log of this run can be found in:

[2019-05-15T08:32:21.831Z] npm ERR!     /root/.npm/_logs/2019-05-15T08_27_12_680Z-debug.log

**Comment by deomrinal** [comment]
Facing the same issue in Linux as well as windows.

**Comment by shmulik311** [comment]
same issue 

[root@LINX102083 test]# node -v
v8.4.0
[root@LINX102083 test]# npm -v
5.6.0
OS - Red Hat Enterprise Linux Server release 7.5 (Maipo)

[root@LINX102084TDT ~]# npm i node-rdkafka

> node-rdkafka@2.6.1 install /root/node_modules/node-rdkafka
> node-gyp rebuild

gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: This is most likely not a problem with node-gyp or the package itself and
gyp ERR! stack is related to network connectivity. In most cases you are behind a proxy or have bad
gyp ERR! stack network settings.
gyp ERR! stack     at Request.<anonymous> (/usr/lib/node_modules/npm/node_modules/node-gyp/lib/install.js:193:21)
gyp ERR! stack     at emitOne (events.js:115:13)
gyp ERR! stack     at Request.emit (events.js:210:7)
gyp ERR! stack     at Request.onRequestError (/usr/lib/node_modules/npm/node_modules/request/request.js:884:8)
gyp ERR! stack     at emitOne (events.js:115:13)
gyp ERR! stack     at ClientRequest.emit (events.js:210:7)
gyp ERR! stack     at TLSSocket.socketErrorListener (_http_client.js:401:9)
gyp ERR! stack     at emitOne (events.js:115:13)
gyp ERR! stack     at TLSSocket.emit (events.js:210:7)
gyp ERR! stack     at emitErrorNT (internal/streams/destroy.js:64:8)
gyp ERR! System Linux 3.10.0-862.14.4.el7.x86_64
gyp ERR! command "/usr/bin/node" "/usr/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /root/node_modules/node-rdkafka
gyp ERR! node -v v8.4.0
gyp ERR! node-gyp -v v3.6.2
gyp ERR! not ok
npm WARN root@1.0.0 No description
npm WARN root@1.0.0 No repository field.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-rdkafka@2.6.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-rdkafka@2.6.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-05-19T08_02_22_079Z-debug.log

**Comment by deomrinal** [comment]
Can someone provide any resolution to this as we are unable to progress further.

**Comment by nfmelendez** [comment]
same is happening to me

**Comment by ankon** [comment]
@deomrinal 

> `[2019-05-15T08:27:12.102Z] env: can't execute 'bash': No such file or directory`

Can you verify that you have bash installed?

---
@shmulik311 this is a similar error, but likely completely unrelated. As the message says, 
> gyp ERR! stack Error: This is most likely not a problem with node-gyp or the package itself and
> gyp ERR! stack is related to network connectivity. In most cases you are behind a proxy or have bad
> gyp ERR! stack network settings.

Check your network connectivity.

**Comment by deomrinal** [comment]
@ankon  I have bash installed in my system.
@shmulik311 Network connectivity there.

**Comment by sidharthv96** [comment]
I'm also having the same issue in Windows.
```
> node-rdkafka@2.6.1 install <home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka
> node-gyp rebuild


<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka>if not defined npm_config_node_gyp (node "<home>\AppData\Roaming\nvm\v10.16.0\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "<home>\AppData\Roaming\nvm\v10.16.0\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
Building the projects in this solution one at a time. To enable parallel build, please add the "/m" switch.
  Getting librdkafka from nuget
cl : Command line warning D9025: overriding '/GR-' with '/GR' [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
cl : Command line warning D9025: overriding '/GR-' with '/GR' [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
cl : Command line warning D9025: overriding '/GR-' with '/GR' [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
cl : Command line warning D9025: overriding '/GR-' with '/GR' [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  binding.cc
  callbacks.cc
  common.cc
cl : Command line warning D9025: overriding '/GR-' with '/GR' [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  config.cc
..\src\callbacks.cc(103): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxp
roj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\callbacks.cc(275): error C2589: '(': illegal token on right side of '::' [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
..\src\callbacks.cc(275): error C2059: syntax error: '::' [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
..\src\callbacks.cc(575): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxp
roj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\callbacks.cc(612): warning C4996: 'Nan::Callback::operator ()': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkaf
ka.vcxproj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1638): note: see declaration of 'Nan::Callback::operator ()'
  connection.cc
  errors.cc
  kafka-consumer.cc
  producer.cc
<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\src/workers.h(42): warning C4996: 'Nan::Callback::Call': was declared deprecated (compiling source file ..\src\conn
ection.cc) [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call' (compiling source file ..\src\connection.cc)
<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\src/workers.h(42): warning C4996: 'Nan::Callback::Call': was declared deprecated (compiling source file ..\src\kafk
a-consumer.cc) [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call' (compiling source file ..\src\kafka-consumer.cc)
..\src\kafka-consumer.cc(560): warning C4996: 'v8::Value::ToObject': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka
.vcxproj]
  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
..\src\kafka-consumer.cc(567): warning C4996: 'v8::Value::ToObject': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka
.vcxproj]
  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
  topic.cc
  workers.cc
<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\src/workers.h(42): warning C4996: 'Nan::Callback::Call': was declared deprecated (compiling source file ..\src\prod
ucer.cc) [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call' (compiling source file ..\src\producer.cc)
..\src\producer.cc(115): warning C4996: 'v8::Value::ToObject': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpr
oj]
  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
..\src\producer.cc(122): warning C4996: 'v8::Value::ToObject': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpr
oj]
  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
..\src\producer.cc(373): warning C4996: 'v8::Value::ToObject': was declared deprecatedadmin.cc [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkaf
ka.vcxproj]

  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
..\src\producer.cc(396): warning C4996: 'v8::Value::ToObject': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpr
oj]
  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
  win_delay_load_hook.cc
..\src\topic.cc(118): warning C4996: 'v8::Value::ToObject': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\src/workers.h(42): warning C4996: 'Nan::Callback::Call': was declared deprecated (compiling source file ..\src\work
ers.cc) [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call' (compiling source file ..\src\workers.cc)
..\src\workers.cc(69): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj
]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(78): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj
]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(116): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(127): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(174): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(183): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(223): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(232): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(261): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(301): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(342): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(351): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(386): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(397): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(446): warning C4996: '_sleep': This function or variable has been superceded by newer library or operating system functionality. Consider using Sleep instead. See online help for deta
ils. [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt\stdlib.h(1274): note: see declaration of '_sleep'
..\src\workers.cc(457): warning C4996: '_sleep': This function or variable has been superceded by newer library or operating system functionality. Consider using Sleep instead. See online help for deta
ils. [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  C:\Program Files (x86)\Windows Kits\10\Include\10.0.17763.0\ucrt\stdlib.h(1274): note: see declaration of '_sleep'
..\src\workers.cc(484): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(498): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(580): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(597): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(644): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(653): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(696): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(705): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(758): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(767): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(805): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(814): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'<home>\dev\webhooks-service\lifecycle-service\nod
  e_modules\node-rdkafka\src/workers.h(42): warning C4996: 'Nan::Callback::Call': was declared deprecated (compiling source file ..\src\admin.cc)

  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call' (compiling source file ..\src\admin.cc)..\src\workers.cc(852):
   warning C4996: 'Nan::Callback::Call': was declared deprecated

  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(861): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
..\src\workers.cc(900): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
..\src\admin.cc(111): warning C4996: 'v8::Value::ToObject': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxproj]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
  <home>\.node-gyp\10.16.0\include\node\v8.h(10046): note: see declaration of 'v8::Value::ToObject'
..\src\workers.cc(909): warning C4996: 'Nan::Callback::Call': was declared deprecated [<home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka\build\node-librdkafka.vcxpro
j]
  <home>\dev\webhooks-service\lifecycle-service\node_modules\nan\nan.h(1740): note: see declaration of 'Nan::Callback::Call'
gyp ERR! build error
gyp ERR! stack Error: `C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\MSBuild\15.0\Bin\MSBuild.exe` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onExit (<home>\AppData\Roaming\nvm\v10.16.0\node_modules\npm\node_modules\node-gyp\lib\build.js:262:23)
gyp ERR! stack     at ChildProcess.emit (events.js:198:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:248:12)
gyp ERR! System Windows_NT 10.0.17763
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "<home>\\AppData\\Roaming\\nvm\\v10.16.0\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd <home>\dev\webhooks-service\lifecycle-service\node_modules\node-rdkafka
gyp ERR! node -v v10.16.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
npm WARN acorn-dynamic-import@4.0.0 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-rdkafka@2.6.1 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-rdkafka@2.6.1 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by Aditya-Viswanadha** [comment]
Same issue.

`
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (../../)
gyp ERR! stack     at ChildProcess.emit (events.js:200:13)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:272:12)
gyp ERR! System Darwin 18.6.0
gyp ERR! command "/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd ../node_modules/node-rdkafka
gyp ERR! node -v v12.4.0
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: google-closure-compiler-linux@20190618.0.0 (node_modules/google-closure-compiler-linux):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for google-closure-compiler-linux@20190618.0.0: wanted {"os":"linux","arch":"x64,x86"} (current: {"os":"darwin","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: google-closure-compiler-windows@20190618.0.0 (node_modules/google-closure-compiler-windows):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for google-closure-compiler-windows@20190618.0.0: wanted {"os":"win32","arch":"x64,x86"} (current: {"os":"darwin","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-rdkafka@2.7.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the node-rdkafka@2.7.0 install script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

`

**Comment by d-sphinx** [comment]
Same errors. 2.7.0 doesn't work either.

**Comment by smile-eh** [comment]
Hi, I had issues using the latest version (12.7.0) of node and was able to fix it by changing to the LTS version (10.16.1).

This may be obvious for some but I figured I'd report back since this page is high on google. I did not have issues with 8.2.1 or 8.4.0 as reported above, so I am unable to assist with that.

**Comment by Aditya-Viswanadha** [comment]
Any update on the fix for this?

**Comment by raitucarp** [comment]
using the latest version of node, I confirm that also encounter this issue when installing

**Comment by yrsh** [comment]
Same error. 
node-rdkafka: 2.8.0
node: 11.15.0

**Comment by sam-github** [comment]
Issues above seem to be a variety of stuff, not all related. Note the node versions.

v12.x isn't supported, see #733, which was fixed in #759 but not released yet.

The problems on node 10.x are some other cause, likely local config related.

**Comment by iradul** [comment]
New version `2.8.0` was released few days ago with #759 merged ;)
> Issues above seem to be a variety of stuff, not all related.

Exactly. Windows users should try with Visual Studio command prompt instead of the built-in one.

**Comment by sam-github** [comment]
Apologies @iradul I didn't notice the commit landed with a different hash.
